### PR TITLE
Fix float export precision

### PR DESCRIPTION
**Pull request by mvorisek**
related/needed for https://github.com/sebastianbergmann/phpunit/pull/4972

also add tests for NAN and microoptimize bool dump

### CODE REVIEW

File: tests/ExporterTest.php [hunk 1]
```diff
@@ -52,6 +52,13 @@ public function exportProvider()
             'export int 1'                  => [1, '1'],
             'export float 1.0'              => [1.0, '1.0'],
             'export float 1.2'              => [1.2, '1.2'],
+            'export float 1 / 3'            => [1 / 3, '0.3333333333333333'],
+            'export float 1 - 2 / 3'        => [1 - 2 / 3, '0.33333333333333337'],
```
Comment (mvorisek) [inline tests/ExporterTest.php:56]: previously `0.3333333333333333` was dumped (as for `1 / 3`, but the stored value is slighly different)

**Comment by mvorisek** [comment]
After a merge, please tag a new 3.1 and 4.0 release.

**Comment by mvorisek** [comment]
Hi @sebastianbergmann, can you please look into this PR? For me, the limited exported float precision is a clear bug. a) the exported precision should not depends on php.ini/local configuration, b) full precision should always be exported, otherwise two export numbers can be the same for different values.

**Comment by mvorisek** [comment]
@sebastianbergmann can I please ask you to merge this bugfix PR and tag a new release?

**Comment by sebastianbergmann** [comment]
I am sorry, but "also add tests for NAN and microoptimize bool dump" throws me off. Please propose these changes in separate pull requests.

**Comment by mvorisek** [comment]
> I am sorry, but "also add tests for NAN and microoptimize bool dump" throws me off. Please propose these changes in separate pull requests.

done in https://github.com/sebastianbergmann/exporter/pull/48

NAN/INF - this PR is about exporting full precision of `float` type and NAN/INF tests are related, is it ok to keep them in this PR?

**Comment by sebastianbergmann** [comment]
Thank you.

Versions 3.1.5 and 4.0.5 have been tagged, https://github.com/sebastianbergmann/phpunit/pull/4972 can be updated.

**Pull request merged**